app/src/main/java/com/coopra/nebulus/views/adapters/AlternativeAdapter.java

```diff
@@ -0,0 +1,79 @@
+package com.coopra.nebulus.views.adapters;
+
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.coopra.data.Track;
+import com.coopra.nebulus.R;
+
+import java.util.List;
+import java.util.Locale;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+public class AlternativeAdapter extends RecyclerView.Adapter<AlternativeAdapter.MyViewHolder> {
+    private List<Track> mTracks;
```

Comment:
Allocate this immediately.

```suggestion
    private final List<Track> mTracks = new ArrayList<>();
```